File: src/main/java/leetcode/CountingBits_338.java
package leetcode;

public class CountingBits_338 {

	public int[] countBits(int num) {
		int res[] = new int[num + 1];
		int i = res.length;

		while (num > 0) {
			int temp = num;

			int count = oneCount(temp);
			i--;
			res[i] = count;
			num--;
		}

		return res;
	}

	private int oneCount(int n) {
		int res = 0;

		while (n > 0) {
			if ((n & 1) == 1) res++;
			n = n >> 1;
		}

		return res;
	}
}

File: Java/src/main/java/leetcode/CountingBits_338.c
#include<stdio.h>

int main(void) {
    int n = 5;
    countBits(n, sizeof(n));
}

int* countBits(int num, int* returnSize) {
    int *res = malloc(returnSize);
    int i = returnSize;

    while (num > 0) {
        int temp = num, count = 0;

        while (temp > 0) {
            if ((temp & 1) == 1) count++;
            temp = temp >> 1;
        }
        i--;
        res[i] = count;
        num--;
    }

    return *res;
}